### ISSUE TITLE
Inline styles not added for blocks with custom font

### ISSUE THREAD

**Issue by swissspidy** [ampproject/amp-wp#3344]
**Steps to reproduce**

1. Create new text block with some content
2. Select `Georgia` as custom font
3. Preview story
4. See that text is not displayed in `Georgia` font.

Works fine when using a font from Google Fonts, but not for ones provided natively by the OS / browser.

Likely introduced in #3328.

**Comment by swissspidy** [comment]
The problem seems to be that `\AMP_Story_Post_Type::render_block_with_google_fonts()` bails early when a font doesn't have a `$font['handle']` (which is the case if it's not a Google font).

Either we can still enqueue the CSS there for non-Google Fonts, or we have to load those earlier again in `\AMP_Story_Post_Type::enqueue_general_styles` like so:

```diff
--- includes/class-amp-story-post-type.php
+++ includes/class-amp-story-post-type.php
@@ -839,6 +839,18 @@ class AMP_Story_Post_Type {
 		);
 
 		wp_styles()->add_data( self::AMP_STORIES_STYLE_HANDLE, 'rtl', 'replace' );
+
+		$fonts = self::get_fonts();
+		foreach ( $fonts as $font ) {
+			if ( isset( $font['handle'] ) ) {
+				continue;
+			}
+
+			wp_add_inline_style(
+				self::AMP_STORIES_STYLE_HANDLE,
+				self::get_inline_font_style_rule( $font )
+			);
+		}
 	}
 
 	/**
```

**Comment by westonruter** [comment]
We'd still need to prevent adding rules for fonts that aren't being used in the story. So I think `\AMP_Story_Post_Type::render_block_with_google_fonts()` should be modified something like the following:

```diff
diff --git a/includes/class-amp-story-post-type.php b/includes/class-amp-story-post-type.php
index 31737b8d..5d72c9e0 100644
--- a/includes/class-amp-story-post-type.php
+++ b/includes/class-amp-story-post-type.php
@@ -1396,9 +1396,20 @@ class AMP_Story_Post_Type {
 			return $block_content;
 		}
 
-		// Short-circuit if there is no Google Font or the font is already enqueued.
 		$font = self::get_font( $block['attrs'][ $font_family_attribute ] );
-		if ( ! isset( $font['handle'], $font['src'] ) || ! $font || wp_style_is( $font['handle'] ) ) {
+
+		// Add style rule for system fonts.
+		if ( ! isset( $font['handle'], $font['src'] ) ) {
+			// @todo Prevent adding duplicates!
+			wp_add_inline_style(
+				self::AMP_STORIES_STYLE_HANDLE,
+				self::get_inline_font_style_rule( $font )
+			);
+			return $block_content;
+		}
+
+		// Short-circuit if there is no Google Font or the font is already enqueued.
+		if ( wp_style_is( $font['handle'] ) ) {
 			return $block_content;
 		}
 ```

**Comment by swissspidy** [comment]
I briefly tried that method with 

```
wp_add_inline_style(
	self::AMP_STORIES_STYLE_HANDLE,
	self::get_inline_font_style_rule( $font )
);
```

as well before opening the issue but somehow it didn't work (no CSS was output). Presumably because the stylesheet was already printed.

**Comment by swissspidy** [comment]
Potential fix: #3345

**Comment by swissspidy** [comment]
**Testing Instructions**

1. Create a new story
1. Create a new text block and choose "Georgia" as the custom font
1. Create a second text block and choose "Aref Ruqaa" as the custom font
1. Create a third text block and choose "Georgia" as the custom font
1. Create a fourth text block and choose "Arial" as the custom font
1. Create a fifth text block and choose "Aref Ruqaa" as the custom font
1. Save story
1. Preview story
1. Verify that the texts are displayed in their respective font
1. View page source and verify that only 1 stylesheet is loaded from Google Fonts
1. Verify that `[data-font-family="Aref Ruqaa"]{...}` exists only once in the source
1. Verify that `[data-font-family="Georgia"]{...}` exists only once in the source
1. Verify that `[data-font-family="Arial"]{...}` exists only once in the source

**Comment by csossi** [comment]
Verified in QA
